### PR TITLE
Input checking

### PR DESCRIPTION
**Pull request by PowerfulBacon**
Circuit reading from user only allows:
 - Words
 - Spaces
 - Numbers
 - `{}`
 - `:`
 - `,`
 - `.`
 - `&`
 - `*`
 - `=`
 - `()`
 - `'`
 - `"`
 - `#`
Anything else gets replaced with nothing

### CODE REVIEW

File: code/modules/integrated_electronics/core/saved_circuits.dm [hunk 1]
```diff
@@ -232,6 +232,9 @@
 
 	var/error
 
+	//Check for bad inputs
+	var/regex/bad_regex = new("\[^\\w{}:\",.\\-*&+ =\'\\(\\)\\\[\\\]#\]", "gi")
+	bad_regex.Replace(bad_regex, "")
```
Comment (Crossedfall) [inline code/modules/integrated_electronics/core/saved_circuits.dm]: This doesn't do anything

**Review by Crossedfall — DISMISSED** [review]

File: code/modules/integrated_electronics/core/saved_circuits.dm (hunk 1 above)
Comment (park66665) [reply code/modules/integrated_electronics/core/saved_circuits.dm]: The thing should probably be put somewhere here:
https://github.com/BeeStation/BeeStation-Hornet/blob/69d83b6d9aaa9e55061f84b9efb9ba70748f316b/code/modules/integrated_electronics/core/printer.dm#L265-L272

File: code/modules/integrated_electronics/core/saved_circuits.dm [hunk 2]
```diff
@@ -232,6 +232,9 @@
 
 	var/error
 
+	//Check for bad inputs
+	var/regex/bad_regex = new("\[^\\w{}:\",.\\-*&+ =\'\\(\\)\\\[\\\]#\]", "gi")
```
Comment (ike709) [inline code/modules/integrated_electronics/core/saved_circuits.dm]: This regex should be generated once on subsystem init and stored on the subsystem.

File: code/modules/integrated_electronics/core/saved_circuits.dm (hunk 1 above)
Comment (PowerfulBacon) [reply code/modules/integrated_electronics/core/saved_circuits.dm]: The regex runs on the regex itself.

**Comment by PowerfulBacon** [comment]
There we go, actually works now

**Comment by ike709** [comment]
@park66665

**Comment by park66665** [comment]
p sure it works and Bacon actually tested it but I'll test it bc better
safe than sorry eh?

**Comment by park66665** [comment]
I tested it and the regex works now

**Pull request merged**